### ISSUE TITLE
Tracking of customer account pages not working

### ISSUE THREAD

**Issue by DrewML** [magento/m2-devtools#45]
## This issue is a:

- [X] Bug
- [ ] Feature suggestion
- [ ] Documentation issue
- [ ] Other (Please Specify)

### Environment

| Question                    |    Answer     |
| --------------------------- | :-----------: |
| Magento version             | 2.3.0 |
| Browser + version           | N/A |
| node.js version (`node -v`) | N/A |
| npm version (`npm -v`)      | N/A |

## Description

When clicking around in "Record" mode for the RequireJS Bundle Config generator, results are not captured for customer account pages (reported by @davidverholen).

This is very likely the result of [a heuristic](https://github.com/magento/m2-devtools/blob/56a0165644b41cd620928e6b8b62931e52e734c2/src/interop.ts#L82-L88) that is not 100% correct, and we [do not surface that error](https://github.com/magento/m2-devtools/blob/a103e8a0e99cc6ce54671b2bef176e84c62a1db3/src/RequireJS/BundleGenerator/PageCollector.ts#L29-L31) anywhere in the UI.

## Expected result:

The fix for this should be to extend `_getPageConfigType` in `interop.ts` to have a more reliable heuristic.

**Note**: A hard requirement of fixing this will be that it does not require installing any additional modules. One of the goals of the Bundle Generator is to be able to generate config with 0 changes to the Magento store.

## Possible solutions:

We can add in some additional heuristics for pages that don't match the pattern being looked for in `_getPageConfigType`. An example would be a hardcoded list of pages that we can't find the layout handle for, and some additional checks to figure them out.

**Comment by davidverholen** [comment]
it would be possible of course to create something reliable in the magento module that the extension could use.

But this would mean that you then need the extension installed to generate the bundles.
currently it's only a convenient way to load the bundles (mostly)

**Comment by DrewML** [comment]
> But this would mean that you then need the magento module installed to generate the bundles, which currently is not required

Yeah, I'm trying to avoid that as hard as possible. I want bundling to be as simple as possible, and being able to generate a config on a base install with no extra modules is appealing.

I'm working on a fix for this locally. Likely will take some iterations to get the heuristics right

**Comment by davidverholen** [comment]
one counter example regarding the needed bundles of a page (during bundle config generation):
The ce customer account mostly is not rendered using ui components so also mostly needs the shared bundle.

In contrary, in ee, for example, the requisition lists or the company account acls have it's completely own set of js components, which then would also be included into the shared or customer_account bundle.

So if you would bundle all js components for the customer account into one bundle, there would still be a pretty large overhead for something like customer_account_dashboard.

The same of course could happen for custom pages to an uncontrollable extend

**Comment by davidverholen** [comment]
really not sure about it / would need a bit of testing / investigation: Would it maybe make sense, to create the bundle names based on a combination of loaded layout handles / body classes? E.g. md5 hash of the body classes and then load them based on the applied layout handles

**Comment by davidverholen** [comment]
just maybe (to continue speaking with myself here ;) ) build something like @DrewML s solution as a default, but add configurations to the browser extension to enable the user to customize the bundle configuration.

For example bundling all customer-account pages into one like @DrewML suggested. But adding a checkbox to divide it into multiple bundles with the combined layout handle approach.

The magento extension could then have a fallback mechanism. Since we hopefully all have the full page cache enabled in production, it should not really have a noticeable impact on performance testing for the different bundles.

**Comment by DrewML** [comment]
(Haven't forgotten about this, just bogged down with some other work atm. Will get back into this one soon)

**Comment by davidverholen** [comment]
As another point pro many small bundles: 
overall performance of a page will probably be better with larger reusable bundles by leveraging browser caching.
From a SEO perspective, since google page speed insights (as far as I know) always only evaluates one page at a time, the smallest possible bundle size would be optimal for that (so no shared bundle at all as the absolute extreme)

**Comment by DrewML** [comment]
I'm working on a prototype that may solve this problem in a bit of a different way, but would yield more control over to developers. Will share when I have more details
